src/models/data.py

```diff
@@ -545,6 +558,17 @@ def __next__(self) -> Tuple[Tuple[Union[np.ndarray, torch.Tensor],
             final_x_ym = np.concatenate(out_x_ym, axis=0)
             final_y = np.concatenate(out_y, axis=0)
 
+            if self.shuffle_data:
```

Comment:
can you talk me through the shuffling...

This is shuffling the years? So we see `1981-01` first but then `1999-05` next?